qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/CreateAffiliationView.groovy

```diff
@@ -50,22 +51,43 @@ class CreateAffiliationView extends VerticalLayout {
     }
 
     private void initLayout() {
-        this.organisationField = new TextField("Organisation Name")
-        organisationField.setPlaceholder("name of the organisation")
-        organisationField.setDescription("Please enter the name of the organisation e.g. Universität Tübingen.")
-
-        this.addressAdditionField = new TextField("Department")
-        addressAdditionField.setPlaceholder("optional department name")
+        this.organisationBox = new ComboBox<>("Organisation Name")
+        
+        // we don't need the whole affiliation object, just the unique organization names.
+        List<String> organisationNames = sharedViewModel.affiliations.stream().map(affi -> affi.organisation).distinct().collect(Collectors.toList())
+
+        organisationBox.setItems(organisationNames)
+        organisationBox.setTextInputAllowed(true)
```

Comment:
With `setItems` the data is thrown away and a new data provider is created. This way we cannot update and add new entries when the list is modified.

Reply:
The new entries come from the sharedViewModel, which is updated/reloaded upon registration if I understand correctly.